azkaban-common/src/main/java/azkaban/executor/OnContainerizedExecutionEventListener.java

```diff
@@ -53,13 +56,39 @@ void restartExecutableFlow(final ExecutableFlow originalExFlow) {
       logger.error(e.getMessage());
       return;
     }
-    final ExecutableFlow retryExFlow =
+    ExecutableFlow retryExFlow =
         this.executorManagerAdapter.createExecutableFlow(project, flow);
+
+    final ExecutionOptions options = originalExFlow.getExecutionOptions();
+
+    final String restartStrategy = options.getFlowParameters().
```

Comment:
WDYT about converting the string value into a FlowRetryStrategy value and work with the enum type instead of a string type?